sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Processor/EventProcessorClientTests.cs

```diff
@@ -700,113 +674,1998 @@ public async Task IsRunningReturnsTrueWhileStopProcessingAsyncIsNotCalled()
+                .Setup(consumer => consumer.ReadEventsFromPartitionAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<EventPosition>(),
+                    It.IsAny<ReadEventOptions>(),
+                    It.IsAny<CancellationToken>()))
+                .Callback(() =>
+                {
+                    if (Interlocked.Increment(ref partitionsBeingProcessed) == partitionIds.Length)
+                    {
+                        completionSource.SetResult(true);
+                    }
+                })
+                .Returns<string, EventPosition, ReadEventOptions, CancellationToken>((partition, position, options, token) =>
+                    MockEndlessPartitionEventEnumerable(options.MaximumWaitTime, token));
+
+            var hasReceivedEventsWhileNotRunning = false;
+
+            // Use the process event handler to track events received after the processor has stopped.
+
+            mockProcessor.ProcessEventAsync += eventArgs =>
+            {
+                if (!mockProcessor.IsRunning)
+                {
+                    hasReceivedEventsWhileNotRunning = true;
+                }
+
+                return Task.CompletedTask;
+            };
+
+            mockProcessor.ProcessErrorAsync += eventArgs => Task.CompletedTask;
+
+            // Establish timed cancellation to ensure that the test doesn't hang.
+
+            using var cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+            await mockProcessor.StartProcessingAsync(cancellationSource.Token);
+
+            while (!completionSource.Task.IsCompleted
+                && !cancellationSource.IsCancellationRequested)
+            {
+                await Task.Delay(25);
+            }
+
+            await mockProcessor.StopProcessingAsync(cancellationSource.Token);
+
+            Assert.That(cancellationSource.IsCancellationRequested, Is.False, "The processor should have stopped without cancellation.");
+
+            // Wait a bit to give the stopped processor the chance to trigger the event handler. This scenario is not expected.
+
+            await Task.Delay(5 * maximumWaitTimeInMilli);
+
+            Assert.That(hasReceivedEventsWhileNotRunning, Is.False, "The processor should not have received events while not running.");
+        }
+
+        /// <summary>
+        ///   Verifies functionality of the <see cref="EventProcessorClient.StopProcessingAsync" />
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public async Task StopProcessingAsyncShouldSurfaceLoadBalancingException()
+        {
+            var mockProcessor = new Mock<EventProcessorClient>(Mock.Of<StorageManager>(), "consumerGroup", "namespace", "eventHub", Mock.Of<Func<EventHubConnection>>(), default, default) { CallBase = true };
+            var completionSource = new TaskCompletionSource<bool>();
+
+            mockProcessor
```

Comment:
Is that considered a load balancing exception or is the name a bit misleading?

Reply:
The name is misleading. I didn't have the `PartitionLoadBalancer` in mind when I wrote this method. What do you think about `BackgroundRunningTaskException`?

Reply:
Works for me.   I think it better represents the scenario.